addon/services/scroll-activity.js

```diff
@@ -17,7 +16,6 @@ const SCROLL_EVENT_TYPE_VERTICAL = 'vertical';
 const SCROLL_EVENT_TYPE_HORIZONTAL = 'horizontal';
 const SCROLL_EVENT_TYPE_DIAGONAL = 'diagonal';
 
-@classic
 export default class ScrollActivityService extends FastBootAwareEventManagerService {
   init() {
```

Comment:
Doesn't this need to be converted to `constructor` if you're going to remove `@classic`?

Reply:
Hmm, I guess it's still supported for Services for some reason. constructor would be preferred, though.

Reply:
Switching from init() to constructor would break the way the overrides work.
For example `user-idle` service we are supposed to override activeEvents like this:
```
export default class UserIdle extends UserIdleService {
  activeEvents = ['mousedown', 'keydown']
};
```
It will no longer work as the constructor of UserIdleService will have a default value of `activeEvent`, thus override won't have effect for the constructor.
We would need to change the API to make it work with constructors: provide overridable params as constructor arg or make a method, something like getActiveEvents() which can be overwritten.

I wanted to keep this change non-breaking, thus didn't change the init().

Reply:
Makes sense, thanks for the explanation!